### PR TITLE
Palindrome for odd case and setup testing

### PR DESCRIPTION
**Pull request by venomnert**
## Summary
* The previous commit only solved for even number input. Odd number input was not solved for.
* No testing was setup

## Purposed changes
* Odd number input resolves to the next largest palindrome
* Setup test for simple and edge cases.

## How to test
* Run `cd ./palindrome && mix test`

### CODE REVIEW

File: palindrome/lib/palindrome.ex [hunk 1]
```diff
@@ -28,41 +29,72 @@ defmodule Palindrome do
...
+    |> process_input()
   end
 
   defp to_integer(input) when is_list(input) do
     input
     |> List.to_string()
     |> String.to_integer()
   end
+
+  defp is_equal(input) do
```
Comment (narinderberi) [inline palindrome/lib/palindrome.ex:95]: Is `is_equal` checking for equality?

File: palindrome/lib/palindrome.ex [hunk 2]
```diff
@@ -28,41 +29,72 @@ defmodule Palindrome do
...
     input_int = input |> to_integer()
 
     (input_int + :math.pow(10, increment_by))
     |> trunc()
     |> Integer.to_string()
-    |> get_palindrome()
+    |> process_input()
+  end
+
+  def increment(input, :odd) do
```
Comment (narinderberi) [inline palindrome/lib/palindrome.ex:79]: Is this just `increment-ing` the input?

File: palindrome/lib/palindrome.ex [hunk 3]
```diff
@@ -28,41 +29,72 @@ defmodule Palindrome do
...
+    input
+    |> reverse(:odd)
+    |> check(input)
+    |> case do
+      {true, new} -> {:ok, to_integer(new)}
+      {false, new} -> increment(new, :odd)
+    end
+  end
+
+  def reverse(input, :even) do
```
Comment (narinderberi) [inline palindrome/lib/palindrome.ex:50]: This is not simply `reverse-ing` the input right? You intend to convey something else.

File: palindrome/lib/palindrome.ex [hunk 4]
```diff
@@ -28,41 +29,72 @@ defmodule Palindrome do
 
   def compute(input, :even) do
```
Comment (narinderberi) [inline palindrome/lib/palindrome.ex:30]: what does `compute` convey?

File: palindrome/lib/palindrome.ex [hunk 5]
```diff
@@ -1,14 +1,15 @@
...
 
-  # TODO test for valid false input
   def get_palindrome(input) when is_integer(input) do
     input
     |> Integer.to_string()
-    |> get_palindrome()
+    |> process_input()
   end
 
-  def get_palindrome(input) when is_binary(input) do
```
Comment (narinderberi) [inline palindrome/lib/palindrome.ex:11]: We are not `getting palindrome`. We are `finding the next larger palindrome`

File: palindrome/lib/palindrome.ex [hunk 6]
```diff
@@ -1,14 +1,15 @@
 defmodule Palindrome do
   @moduledoc false
 
-  # TODO test for valid false input
   def get_palindrome(input) when is_integer(input) do
     input
     |> Integer.to_string()
-    |> get_palindrome()
+    |> process_input()
```
Comment (narinderberi) [inline palindrome/lib/palindrome.ex:7]: process_input name is extremely generic, not creating clarity on what it does. process_input may mean anything.
